File: backend/app/core/session_store.py
from app.models.session import SessionState

# In-memory store: session_id -> SessionState
_store: dict[str, SessionState] = {}


def get(session_id: str) -> SessionState | None:
    return _store.get(session_id)


def all_sessions() -> list[SessionState]:
    return list(_store.values())


def put(session: SessionState) -> None:
    _store[session.session_id] = session


def delete(session_id: str) -> bool:
    if session_id in _store:
        del _store[session_id]
        return True
    return False


def exists(session_id: str) -> bool:
    return session_id in _store

File: app/backend-bun/src/session-store.ts
export interface SessionState {
  session_id:    string;
  fe_file:       string;
  energy:        number | null;
  area:          number | null;
  scale:         number | null;
  vertex_count:  number | null;
  edge_count:    number | null;
  facet_count:   number | null;
  last_accessed: Date;
}

const store = new Map<string, SessionState>();

function touch(s: SessionState): SessionState {
  s.last_accessed = new Date();
  return s;
}

export function get(sessionId: string): SessionState | undefined {
  const s = store.get(sessionId);
  return s ? touch(s) : undefined;
}

export function put(session: SessionState): void {
  store.set(session.session_id, session);
}

export function del(sessionId: string): boolean {
  return store.delete(sessionId);
}

export function exists(sessionId: string): boolean {
  return store.has(sessionId);
}

export function count(): number {
  return store.size;
}

export function all(): SessionState[] {
  return [...store.values()];
}
